modules/service/import-service.js

```diff
@@ -90,24 +69,40 @@ const constants = {
 };
 Object.freeze(constants);
 
-/**
- *
- */
-class OtJsonImporter {
-    /**
-     * Default constructor. Creates instance of otJsonImporter.
-     * @param ctx IoC context
-     */
+class ImportService {
     constructor(ctx) {
         this.db = ctx.graphStorage;
-        this.log = ctx.logger;
-        this.config = ctx.config;
-        this.notifyError = ctx.notifyError;
+        this.schemaValidator = ctx.schemaValidator;
         this.web3 = ctx.web3;
-
-        this.schemaValidator = new SchemaValidator(ctx);
     }
 
+    async getImport(datasetId, encColor = null) {
+        if (![null, 'red', 'green', 'blue'].includes(encColor)) {
+            throw Error('Invalid encryption color.');
+        }
+        const vertices = await this.db.findVerticesByImportId(datasetId);
+        const edges = await this.db.findEdgesByImportId(datasetId);
+        const metadata = await this.db.findMetadataByImportId(datasetId);
+
+        // TODO: Check if date with specified encryption exists
+        if (encColor != null) {
+            vertices.filter(v => v.encrypted != null)
+                .forEach(v => v.data = v.encrypted[encColor.charAt(0)]);
```

Comment:
edges can also be encrypted, and we also changed how encryption color is stored. But that's outside the scope of this pull request, let's add a different task to refactor this part of the export

Reply:
TODO comment added